internal/command/testdata/test/pass_with_run_block_variable/tests/main.tftest.hcl

```diff
@@ -0,0 +1,16 @@
+# tests/main.tftest.hcl
+
+provider "docker" {
+  host = run.setup.aws_access_key_id
```

Comment:
This is a test for the scenario of allowing `provider` blocks to access `run` outputs. What about the other scenario -  allowing `provider` blocks to access variables?
I'm asking because I didn't see any relevant code change or relevant test.

Reply:
I had a fix for accessing variables in the provider block too, but after the latest pull from the main branch, the variables were working fine in the provider block without any explicit code changes. However, I will add some test cases for reference.